### ISSUE TITLE
Getting MNIST image classifier accuracy, precision and other eval.stats() metrics as NaN !!!!

### ISSUE THREAD

**Issue by smilebuddy** [deeplearning4j/nd4j#2147]
#### Issue Description

I am trying to build image classifier using deeplearning4J in spark cluster. Java code is attached. My code is running successfully over spark cluster using spark-submit command, but the my model evaluation result gives Accuracy as NaN. I earlier thought that it's an issue with the model parameter and i tried tuning hyper-parameters like learning rate(tried 0.01, 0.0001, 0.000001) but gor no success. 
Please tell me what's wrong here? 
 

#### Version Information

Please indicate relevant versions, including, if relevant:

* Deeplearning4j version: 0.9.1
* platform information (OS, etc) : linux, spark over hadoop cluster

[MnistMLPDistributedExample.txt](https://github.com/deeplearning4j/nd4j/files/1320843/MnistMLPDistributedExample.txt)

**Comment by tfedyanin** [comment]
I hava same issue. Can you explain how to fix? 
Now I use next fix:
```java        
// Perform eval (non-distributed)
Evaluation eval = new Evaluation(10);
MultiLayerNetwork model = sparkNet.getNetwork();
testData.collect().forEach(entry -> {
INDArray output = model.output(entry.getFeatureMatrix()); //get the networks prediction
eval.eval(entry.getLabels(), output); //check the prediction against the true class
});
```

**Comment by AlexDBlack** [comment]
@tfedyanin https://github.com/deeplearning4j/dl4j-examples/pull/562

**Comment by tfedyanin** [comment]
@AlexDBlack Thank you for explanation!
